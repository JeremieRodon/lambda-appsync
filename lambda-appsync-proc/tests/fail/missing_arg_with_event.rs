use lambda_appsync::{appsync_lambda_main, appsync_operation, AppsyncError, AppsyncEvent, ID};

appsync_lambda_main!("../../../../schema.graphql");

// Missing argument
#[appsync_operation(mutation(createPlayer), with_appsync_event)]
async fn create_player() -> Result<Player, AppsyncError> {
    Ok(Player {
        id: ID::new(),
        name: "Test".to_string(),
        team: Team::Rust,
    })
}

// Missing op argument
#[appsync_operation(mutation(deletePlayer), with_appsync_event)]
async fn delete_player(_event: &AppsyncEvent<Operation>) -> Result<Player, AppsyncError> {
    Ok(Player {
        id: ID::new(),
        name: "deleted".into(),
        team: Team::MultiWordsTeam,
    })
}

// Missing event argument
#[appsync_operation(query(player), with_appsync_event)]
async fn get_player(_id: ID) -> Result<Option<Player>, AppsyncError> {
    Ok(None)
}
