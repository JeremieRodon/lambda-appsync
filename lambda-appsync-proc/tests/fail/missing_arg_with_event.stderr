error[E0593]: function is expected to take 2 arguments, but it takes 0 arguments
 --> tests/fail/missing_arg_with_event.rs:7:10
  |
6 | #[appsync_operation(mutation(createPlayer), with_appsync_event)]
  | ----------------------------------------------------------------
  | |
  | takes 0 arguments
  | required by a bound introduced by this call
7 | async fn create_player() -> Result<Player, AppsyncError> {
  |          ^^^^^^^^^^^^^ expected function that takes 2 arguments
  |
note: required by a bound in `create_player::with_event::check_signature`
 --> tests/fail/missing_arg_with_event.rs:3:1
  |
3 | appsync_lambda_main!("../../../../schema.graphql");
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `check_signature`
  = note: this error originates in the macro `appsync_lambda_main` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0593]: function is expected to take 2 arguments, but it takes 1 argument
  --> tests/fail/missing_arg_with_event.rs:17:10
   |
16 | #[appsync_operation(mutation(deletePlayer), with_appsync_event)]
   | ----------------------------------------------------------------
   | |
   | takes 1 argument
   | required by a bound introduced by this call
17 | async fn delete_player(_event: &AppsyncEvent<Operation>) -> Result<Player, AppsyncError> {
   |          ^^^^^^^^^^^^^ expected function that takes 2 arguments
   |
note: required by a bound in `delete_player::with_event::check_signature`
  --> tests/fail/missing_arg_with_event.rs:3:1
   |
3  | appsync_lambda_main!("../../../../schema.graphql");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `check_signature`
   = note: this error originates in the macro `appsync_lambda_main` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0593]: function is expected to take 2 arguments, but it takes 1 argument
  --> tests/fail/missing_arg_with_event.rs:27:10
   |
26 | #[appsync_operation(query(player), with_appsync_event)]
   | -------------------------------------------------------
   | |
   | takes 1 argument
   | required by a bound introduced by this call
27 | async fn get_player(_id: ID) -> Result<Option<Player>, AppsyncError> {
   |          ^^^^^^^^^^ expected function that takes 2 arguments
   |
note: required by a bound in `player::with_event::check_signature`
  --> tests/fail/missing_arg_with_event.rs:3:1
   |
3  | appsync_lambda_main!("../../../../schema.graphql");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `check_signature`
   = note: this error originates in the macro `appsync_lambda_main` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/fail/missing_arg_with_event.rs:26:1
   |
26 | #[appsync_operation(query(player), with_appsync_event)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | expected a tuple with 2 elements, found one with 1 element
   | this expression has type `(ID, &AppsyncEvent<Operation>)`
   |
   = note: expected tuple `(ID, &AppsyncEvent<Operation>)`
              found tuple `(_,)`
   = note: this error originates in the attribute macro `appsync_operation` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> tests/fail/missing_arg_with_event.rs:6:1
  |
6 | #[appsync_operation(mutation(createPlayer), with_appsync_event)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | expected `(String, &AppsyncEvent<Operation>)`, found `()`
  | this expression has type `(std::string::String, &AppsyncEvent<Operation>)`
  |
  = note:  expected tuple `(std::string::String, &AppsyncEvent<Operation>)`
          found unit type `()`
  = note: this error originates in the attribute macro `appsync_operation` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/fail/missing_arg_with_event.rs:16:1
   |
16 | #[appsync_operation(mutation(deletePlayer), with_appsync_event)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | expected a tuple with 2 elements, found one with 1 element
   | this expression has type `(ID, &AppsyncEvent<Operation>)`
   |
   = note: expected tuple `(ID, &AppsyncEvent<Operation>)`
              found tuple `(_,)`
   = note: this error originates in the attribute macro `appsync_operation` (in Nightly builds, run with -Z macro-backtrace for more info)
